### PR TITLE
Implement UDP broadcast to make the robot discoverable on the network

### PR DESCRIPTION
**Pull request by miludmann**
The proxy-robots can now find the robots on the network who broadcast a specific message.
Also, naming refactor and misc. cleanup.
#125

### CODE REVIEW

File: robots-ev3-module/src/main/java/orwell/tank/SimpleEscapeKeyListener.java [hunk 1]
```diff
@@ -7,9 +7,9 @@
 import org.slf4j.LoggerFactory;
 
 public class SimpleEscapeKeyListener implements KeyListener {
-    private final static Logger logback = LoggerFactory.getLogger(SimpleEscapeKeyListener.class);
+    private static final Logger logback = LoggerFactory.getLogger(SimpleEscapeKeyListener.class);
 
-    private boolean wasKeyPressed = false;
+    private boolean wasKeyPressed;
```
Comment (dchilot) [inline robots-ev3-module/src/main/java/orwell/tank/SimpleEscapeKeyListener.java]: It is probably better to have the value initialized to false.

File: robots-ev3-module/src/main/java/orwell/tank/actions/Move.java [hunk 2]
```diff
@@ -7,8 +7,8 @@
 import java.util.List;
 
 public class Move implements IInputAction {
-    private final static Logger logback = LoggerFactory.getLogger(Move.class);
-    private boolean hasMove = false;
+    private static final Logger logback = LoggerFactory.getLogger(Move.class);
+    private boolean hasMove;
```
Comment (dchilot) [inline robots-ev3-module/src/main/java/orwell/tank/actions/Move.java]: This is never set to false.

File: robots-ev3-module/src/main/java/orwell/tank/communication/UdpBroadcastDataDecoder.java [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import java.util.Arrays;
 
 public class UdpBroadcastDataDecoder {
-    private final static Logger logback = LoggerFactory.getLogger(UdpBroadcastDataDecoder.class);
+    private static final Logger logback = LoggerFactory.getLogger(UdpBroadcastDataDecoder.class);
     private static final String PROXY_IP_FIELD_NAME = Character.toString('*');
     private boolean isPacketDataCorrect;
```
Comment (dchilot) [inline robots-ev3-module/src/main/java/orwell/tank/communication/UdpBroadcastDataDecoder.java]: This should probably start as false.

File: robots-ev3-module/src/main/java/orwell/tank/communication/UdpProxyFinder.java [hunk 4]
```diff
@@ -7,21 +7,23 @@
...
+    private static final Logger logback = LoggerFactory.getLogger(UdpProxyFinder.class);
     private final int broadcastPort;
     private final DatagramSocket datagramSocket;
     private final UdpBroadcastDataDecoder udpBroadcastDataDecoder;
-    private int attemptsPerformed = 0;
+    private final SimpleEscapeKeyListener simpleEscapeKeyListener;
     private String pushAddress;
     private String pullAddress;
-    private SimpleEscapeKeyListener simpleEscapeKeyListener;
+    private int attemptsPerformed;
```
Comment (dchilot) [inline robots-ev3-module/src/main/java/orwell/tank/communication/UdpProxyFinder.java]: This should be initialized, probably to 0. The value is only incremented and logged though.

File: robots-ev3-module/src/main/java/orwell/tank/config/RobotFileBom.java [hunk 5]
```diff
@@ -94,26 +92,23 @@ public boolean isModelValid() {
                 globalVolume != -1 &&
                 sensorMessageDelayMs != -1 &&
                 leftMortPort != rightMotorPort &&
-                areSensorsPortsDifferent() &&
+               areSensorsPortsDifferent() &&
```
Comment (dchilot) [inline robots-ev3-module/src/main/java/orwell/tank/config/RobotFileBom.java]: There might be an indentation issue.

**Review by dchilot — CHANGES_REQUESTED** [review]

File: robots-ev3-module/src/main/java/orwell/tank/SimpleEscapeKeyListener.java (hunk 1 above)
Comment (miludmann) [reply robots-ev3-module/src/main/java/orwell/tank/SimpleEscapeKeyListener.java]: Ok!

File: robots-ev3-module/src/main/java/orwell/tank/actions/Move.java (hunk 2 above)
Comment (miludmann) [reply robots-ev3-module/src/main/java/orwell/tank/actions/Move.java]: True! Fixed.

File: robots-ev3-module/src/main/java/orwell/tank/communication/UdpBroadcastDataDecoder.java (hunk 3 above)
Comment (miludmann) [reply robots-ev3-module/src/main/java/orwell/tank/communication/UdpBroadcastDataDecoder.java]: Okay. Java default value initialization is false, but if you ask, it's better to explicit it.

File: robots-ev3-module/src/main/java/orwell/tank/communication/UdpProxyFinder.java (hunk 4 above)
Comment (miludmann) [reply robots-ev3-module/src/main/java/orwell/tank/communication/UdpProxyFinder.java]: Same. Fixed!

File: robots-ev3-module/src/main/java/orwell/tank/config/RobotFileBom.java (hunk 5 above)
Comment (miludmann) [reply robots-ev3-module/src/main/java/orwell/tank/config/RobotFileBom.java]: Fixed!

File: robots-ev3-module/src/main/java/orwell/tank/communication/UdpBroadcastDataDecoder.java (hunk 3 above)
Comment (dchilot) [reply robots-ev3-module/src/main/java/orwell/tank/communication/UdpBroadcastDataDecoder.java]: I do not Java enough.

**Review by dchilot — APPROVED** [review]

**Comment by dchilot** [comment]
If the default value is false for boolean and 0 for integer, then it should be fine to rely on those. It is just that I am more used to C++ where those should be taken care of to avoid strange behaviour. So it is up to you to decide if you want to keep with or without the explicit initialisations.

**Pull request merged**